test/README.md

```diff
@@ -0,0 +1,221 @@
+        - Each node should be present in KubeNodeInventory
+    
+
+
+# Ginkgo
+Tests are run using the [Ginkgo](https://onsi.github.io/ginkgo/) test framework. This is built upon the regular go test framework. It's advantages are that it:
+- Has an easily readable test structure using the `Behavior-Driven Development` model that's used in many languages and is applicable outside of GoLang. This model follows a `Given..., When..., Then...` structure. This is implemented in Ginkgo using the `Describe()`, `Context()`, and `It()`/`Specify()` functions. The Ginkgo documentation on [Writing Specs](https://onsi.github.io/ginkgo/#writing-specs) has many examples of this.
+- Utilizes the [Gomega assertion package](https://onsi.github.io/gomega/) for easily understandable test failure errors with the goal that the output will tell you exactly what failed.
+- Has good support for parallelization and structuring which tests should be run in series and which can be run at the same time to speed up the tests.
+- Has extensive documentation and examples from OSS community.
+
+Ginkgo can be used for any tests written in golang, whether they are unit, integration, or e2e tests.
+
+## Bootstrap a Dev Cluster to Run Ginkgo Tests
+- Follow [this](https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-portal?tabs=azure-cli) to create a test cluster and connect to it.
+- Install [Ginkgo](https://onsi.github.io/ginkgo/#getting-started).
+- Navigate to any folder in ./ginkgo-e2e and run command `ginkgo` to trigger the tests on the running cluster.
+- Please note that you don't need testkube to be installed on the cluster to trigger the tests locally on a cluster.
+- You can customize which tests are run with `--label-filter`:
+  - `--label-filter='!/./` is an expression that runs all tests that don't have a label.
+  - `--label-filter='!/./ || LABELNAME` is an expression that runs all tests that don't have a label and tests that have the label `LABELNAME`.
+  - `--label-filter='!(arc-extension,windows)'` is an expression that runs all tests, including those with labels, except for tests labeled `arc-extension` or `windows`.
+- To run only one package of tests, add the path to the tests in the command. For example, to only run the livenessprobe tests on your cluster:
+  ```
+  ginkgo -p -r --keep-going ./livenessprobe
+  ```
+- For more uses of the Ginkgo CLI, refer to the [docs](https://onsi.github.io/ginkgo/#ginkgo-cli-overview).
+
+
+#### Packages
+- [k8s.io/client-go/kubernetes](https://pkg.go.dev/k8s.io/client-go/kubernetes)
+- [k8s.io/api/core/v1](https://pkg.go.dev/k8s.io/api/core/v1)
+- [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go)
+
+# TestKube
+[Testkube](https://docs.testkube.io/) is an OSS runner framework for running the tests inside a Kubernetes cluster. It is deployed as a helm chart on the cluster. Ginkgo is included as one of the out-of-the-box executors supported.
+
```

Comment:
This statement  needs to be rephrased
from
   It is deployed as a helm chart on the cluster
To

It is deployed as a helm release on the cluster.